### PR TITLE
Show which tasks are failing in the status bar

### PR DESCRIPTION
**Pull request by gspencergoog**
When I was gardener last week, I was frustrated several times because Cocoon would say that a task was failing, but all of the tiles on the screen were either green or yellow, so I had to wait a while until something turned red to find out what was broken.

This adds information to the status display that shows which tasks are failing when the build is failing, so that even before any indication is given about the failure, you can tell that it failed.

The main changes in this PR are to add failed task information to the server response for build-status, and to display those failures in the UI in build_dashboard_page.dart. To do this, I created a BuildResponseProto so that it could return more than just a bool.

One alternative to this approach (or an additional thing that could be done) would be to introduce another color for "failed once, but still in progress" tasks (orange?) that would show up right away.

(I originally accidentally submitted a PR off my fork's master, this is now on a branch in my fork. The original was https://github.com/flutter/cocoon/pull/1050)

### CODE REVIEW

**Comment by gspencergoog** [comment]
Still working to address previous comments in https://github.com/flutter/cocoon/pull/1050

**Comment by gspencergoog** [comment]
OK, @CaseyHillers please take a look: this has the failed tasks as part of the build status response.

File: app_dart/lib/src/service/build_status_provider.dart [hunk 1]
```diff
@@ -56,14 +56,15 @@ class BuildStatusService {
       branch: branch,
     ).toList();
     if (statuses.isEmpty) {
-      return BuildStatus.failed;
+      return BuildStatus.success();
```
Comment (CaseyHillers) [inline app_dart/lib/src/service/build_status_provider.dart:59]: Should this still be a failure? If Datastore is failing to return the tasks, this is an infra issue that should close the tree.

**Review by CaseyHillers — APPROVED** [review]
LGTM when comment is addressed

File: app_dart/lib/src/service/build_status_provider.dart (hunk 1 above)
Comment (gspencergoog) [reply app_dart/lib/src/service/build_status_provider.dart:59]: Thanks, that was just an typo.  Fixed.

**Comment by gspencergoog** [comment]
OK, so once this is committed, do I have to deploy it, or will it get deployed automagically?

**Comment by CaseyHillers** [comment]
> OK, so once this is committed, do I have to deploy it, or will it get deployed automagically?

Automagically :-)

The flutter-dashboard GCP project deploys master 8 AM every weekday.

**Comment by CaseyHillers** [comment]
The license check failure seems to be caused by autogenerated files from `flutter build` wiping the licenses. This is probably just a change in behavior where we need to update the gitignore with the latest from flutter/flutter.

**Comment by gspencergoog** [comment]
No, it's because this is the first time that a generated `.pbenum.dart` file has been added. I fixed the licenses script.

**Pull request merged**